### PR TITLE
Refactor codegen plugin

### LINKED ISSUE #3298: Create Go module for codegen plugin
**Describe the feature request**
As we are refactoring the tanzu-framework repository, i would like to create a go module for codegen cli plugin

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by yharish991**
### What this PR does / why we need it
This PR refactors the codegen plugin and makes it a go module

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/3298

### Describe testing done for PR

Ran the tests and codegen plugin separately to verify the functionality

### Release note

```release-note
Refactored codegen plugin and made it a go module
```

### CODE REVIEW

File: cmd/cli/plugin-admin/codegen/go.mod [hunk 1]
```diff
@@ -0,0 +1,92 @@
+module github.com/vmware-tanzu/tanzu-framework/codegen/plugin
+
+go 1.17
+
+replace (
+	github.com/vmware-tanzu/tanzu-framework/apis/cli => ../../../../apis/cli
+	github.com/vmware-tanzu/tanzu-framework/apis/config => ../../../../apis/config
+	github.com/vmware-tanzu/tanzu-framework/cli/runtime => ../../../../cli/runtime
+	k8s.io/apimachinery => k8s.io/apimachinery v0.23.5
```
[the remark this answers is not in the document]
Comment (yharish991) [reply cmd/cli/plugin-admin/codegen/go.mod:9]: no, you don't have too, i was trying to fix the dependencies so had the `apimachinery` in the `replace` directive, will push a fix to remove the `apimachinery` from `replace` directive..

**Review by codegold79 — APPROVED** [review]
LGTM

**Comment by marckhouzam** [comment]
@yharish991 is this ready for review or is it still a WIP?
I am working on refactoring the `builder` plugin and it will help me to base it on your work.

File: pkg/v1/codegen/doc.go [hunk 2]
```diff
@@ -1,5 +0,0 @@
-// Copyright 2021 VMware, Inc. All Rights Reserved.
-// SPDX-License-Identifier: Apache-2.0
-
-// Package codegen provides code generators
-package codegen
```
Comment (marckhouzam) [inline pkg/v1/codegen/doc.go:5]: I don't know what these files are for to be honest, but why are we removing this one?

**Review by marckhouzam — APPROVED** [review]
Nice.
LGTM

File: pkg/v1/codegen/doc.go (hunk 2 above)
Comment (yharish991) [reply pkg/v1/codegen/doc.go:5]: i moved the code from `pkg/v1/codegen/` to `cmd/cli/plugin-admin/code-gen` and we don't a doc.go for main package, so deleted that file

**Comment by yharish991** [comment]
@anujc25 can i get a final review from you on this?

**Comment by marckhouzam** [comment]
As discussed directly with @yharish991, we are pulling out the setting of `buildinfo` for plugins into its own PR: #3345.

This will unblock other PRs that need this change.

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**